servicediscovery/servicediscovery.go

```diff
@@ -55,13 +59,36 @@ func (r Resolver) Resolve() <-chan ResolvedAddresses {
 
 	go func() {
 		for {
-			var addresses []string
+			var (
+				addresses []string
+				uniq      = make(map[string]struct{})
+			)
 			groups := <-manager.SyncCh()
+
 			for _, groupArr := range groups {
 				for i := 0; i < len(groupArr); i++ {
 					group := groupArr[i]
 					for _, target := range group.Targets {
-						addresses = append(addresses, string(target["__address__"]))
+						discoveredLabels := group.Labels.Merge(target)
+						var labelMap = make(map[string]string)
```

Comment:
Maybe its worth to encapsulate this logic in a function to separate concerns

Reply:
hehe sure